extract.py

```diff
@@ -41,7 +42,7 @@ def _run(self):
         try:
             while True:
```

Comment:
why you runnng it with process? should be run as thread
plus
you can't run class method from process